components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -108,8 +111,11 @@ public static void loadInternalApis(String apiFilePath) {
                             if (name == null || name.isEmpty()) {
                                 handleException("Name not specified in one or more handlers");
                             }
+                            // Readiness-probe api is enabled by default.
+                            // other APIs must be enabled manually.
                             if (!Boolean.parseBoolean(
-                                    System.getProperty(Constants.PREFIX_TO_ENABLE_INTERNAL_APIS + name))) {
+                                    System.getProperty(Constants.PREFIX_TO_ENABLE_INTERNAL_APIS + name)) &&
+                                    !READINESS_PROBE_API_NAME.equals(name)) {
```

Comment:
Instead of enabling here what if we have this system property in the start up script. If someone doesn't want to expose this ( as this is over http and have no security ) they can remove the system property in script but if we enable here they won't be able to restrict it easily.

Reply:
Irrespective of the fact that internal APIs are enabled or not, we need to run the readiness probe.

Reply:
yeah but we can make it configurable. That is we can define the system property in startup script without forcing it in the code level so that people who doesn't want to have the probe can stop it.

Reply:
+1 Fixed in 53121f0b023e309f3de70ac5a12dbc95616e479d